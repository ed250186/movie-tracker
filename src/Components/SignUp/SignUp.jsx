
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import './SignUp.scss';
import exit from '../../images/cancel.png'
import React, {Component} from 'react';
import './SignUp.scss';
import { createUser } from '../../apiCalls/apiCalls.js';
import { addUsers } from '../../actions/userActions';

class SignUp extends Component {
  constructor(props) {
    super(props) 
    this.state = {
      name: '',
      email: '',
      password: '',
      userExistsMessage: '',
    }
  }

  handleChange = (e) => {
    const {name, value} = e.target;
    this.setState({[name]: value})
  }

  handleSignUp = async event => {
    event.preventDefault();

    const { name, email, password } = this.state;
    let addUsers = await createUser(name, email, password);
    if (!addUsers.error) {
      this.props.addUsers(addUsers)
      this.props.history.push("/")
    } else {
      this.setState({userExistsMessage: 'Email address already exists in the system. Please log in.'})
    }
    this.resetInputs();
  }

  resetInputs = () => {
    this.setState({name: '', email: '', password: ''});
  };

  render() {
    return (
      <div className='sign-up'>
        <NavLink to='/' className='exit'>
        </NavLink>
        <form onSubmit={this.checkUsers}>
        <NavLink to='/'>
            <img src={exit} alt="exit sign-in button" className='exit-button'/>
          </NavLink>
          <div>
          <input 
            type="text" 
            value={this.state.name}
            placeholder='name' 
            name='name'
            onChange={this.handleChange}
          />
          <input 
            type="text" 
            value={this.state.email}
            placeholder='email' 
            name='email'
            onChange={this.handleChange}
          />
          <input 
            type="password" 
            value={this.state.password}
            placeholder='password' 
            name='password'
            onChange={this.handleChange}
          />
          <input 
            type="submit" 
            name="submit" 
            value='Submit'
            className='button'
            onClick ={event => this.handleSignUp(event)}
            />
            {this.state.userExistsMessage === ''  ? (
              <p></p>
            ) : (
              <p>{this.state.userExistsMessage}</p>
            )}
          </div>
        </form>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  users: state.grabUsers,
})

const mapDispatchToProps = (dispatch) => ({
  addUsers: (users) => dispatch(addUsers(users))
})

export default connect(mapStateToProps, mapDispatchToProps)(SignUp)


