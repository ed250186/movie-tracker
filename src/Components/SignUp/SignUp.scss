.sign-up,
.sign-in {
  align-items: center;
  background: rgba(0, 0, 0, .7);
  cursor: pointer;
  display: flex;
  height: 100%;
  justify-content: center;
  position: absolute;
  width: 100%;
  z-index: 1;
  .exit {
    height: 100%;
    width: 100%;
  }
  form {
    background: #0F1C40;
    height: 50%;
    width: 50%;
    position: absolute;
    z-index: 2;
    div {
      align-content: space-around; 
      display: flex;
      flex-wrap: wrap;
      height: 70%;
      justify-content: center;
      margin: auto;
      width: 70%;
      input {
        border: none;
        border-radius: 50px;
        height: 30px;
        width: 60%;
     }
      ::-webkit-input-placeholder {
        padding-left: 10px;
     }
    }
  }
}
.exit-button {
  height: 20px;
}

p {
  color: white;
}