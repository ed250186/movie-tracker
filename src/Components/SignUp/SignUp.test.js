import React from 'react';
import {shallow} from 'enzyme';
import {SignUp} from './SignUp';

//34.78 |       50 |    22.22 |     38.1
//... 35,37,41,80,99


describe('SignUp', () => {
  let wrapper
  let mockEvent = { 
    preventDefault: jest.fn(),
    target: {
    name: 'email',
    value: 'nathan.froeh@gmail.com'
  }}
  beforeEach(() => {
    wrapper = shallow(<SignUp />);
  })

  it('should match snapshot', () => {
    wrapper.setState({userExistsMessage: ''})
    expect(wrapper).toMatchSnapshot()
  })

  it('should render message', () => {
    wrapper.setState({userExistsMessage: 'user'})
    expect(wrapper).toMatchSnapshot()
  })

  it('should set state', () => {
    expect(wrapper.state('email')).toEqual('')
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('email')).toEqual('nathan.froeh@gmail.com')
  })

  it('should reset state', () => {
    wrapper.setState({
      name: '1', email: '1', password: '1'})
    expect(wrapper.state('name')).toEqual('1')
    expect(wrapper.state('email')).toEqual('1')
    expect(wrapper.state('password')).toEqual('1')
    wrapper.instance().resetInputs()
    expect(wrapper.state('name')).toEqual('')
    expect(wrapper.state('email')).toEqual('')
    expect(wrapper.state('password')).toEqual('')
  })

})