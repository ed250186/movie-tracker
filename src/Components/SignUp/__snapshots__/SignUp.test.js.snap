// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="sign-up"
>
  <NavLink
    className="exit"
    to="/"
  />
  <form>
    <NavLink
      to="/"
    >
      <img
        alt="exit sign-in button"
        className="exit-button"
        src="cancel.png"
      />
    </NavLink>
    <div>
      <input
        name="name"
        onChange={[Function]}
        placeholder="name"
        type="text"
        value=""
      />
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="text"
        value=""
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
      <input
        className="button"
        name="submit"
        onClick={[Function]}
        type="submit"
        value="Submit"
      />
      <p />
    </div>
  </form>
</div>
`;
